python/tests/test_production_server.py

```diff
@@ -0,0 +1,165 @@
+        "test-workspace-1",
+        "test-workspace-with-dashes",
+        "test_workspace_with_underscores",
+    ]
+
+    for workspace in test_workspaces:
+        jwt = await s2iam.get_jwt_database(workspace_group_id=workspace, timeout=30.0)
+
+        assert jwt is not None, f"Database JWT should be generated for workspace {workspace}"
+        assert len(jwt) > 0, f"Database JWT should not be empty for workspace {workspace}"
+        assert jwt.startswith(
+            "eyJ"
+        ), f"Database JWT should be valid JWT format for workspace {workspace}"
+
+        print(f"✓ Successfully generated JWT for workspace: {workspace}")
+
+
+@pytest.mark.asyncio
+async def test_production_server_error_handling():
+    """Test error handling with production server."""
+    client = await require_cloud_role()
+
+    # Skip GCP due to audience issues
+    if client.get_type() == CloudProviderType.GCP:
+        pytest.skip(
+            "GCP temporarily skipped - production server audience configuration in progress"
+        )
+
+    print(f"✓ Detected provider: {client.get_type().value}")
+
+    # Test with invalid workspace ID (should still work but may have different behavior)
+    try:
+        jwt = await s2iam.get_jwt_database(
+            workspace_group_id="", timeout=30.0  # Empty workspace ID
+        )
+        # If this succeeds, validate the JWT
+        if jwt:
+            assert jwt.startswith("eyJ"), "JWT should be valid format even with empty workspace"
+            print("✓ Production server handles empty workspace ID gracefully")
+    except Exception as e:
+        # If this fails, it's also acceptable - just log it
+        print(f"✓ Production server properly rejects empty workspace ID: {e}")
+
+    # Test with very long workspace ID
+    try:
+        long_workspace = "a" * 200  # Very long workspace ID
+        jwt = await s2iam.get_jwt_database(workspace_group_id=long_workspace, timeout=30.0)
+        if jwt:
+            assert jwt.startswith("eyJ"), "JWT should be valid format even with long workspace"
+            print("✓ Production server handles long workspace ID gracefully")
+    except Exception as e:
+        # If this fails, it's also acceptable - just log it
+        print(f"✓ Production server properly handles long workspace ID: {e}")
+
+
+@pytest.mark.asyncio
+async def test_production_server_timeout_handling():
+    """Test timeout handling with production server."""
+    client = await require_cloud_role()
+
+    # Skip GCP due to audience issues
+    if client.get_type() == CloudProviderType.GCP:
+        pytest.skip(
+            "GCP temporarily skipped - production server audience configuration in progress"
+        )
+
+    print(f"✓ Detected provider: {client.get_type().value}")
+
+    # Test with reasonable timeout
+    start_time = __import__("time").time()
+    jwt = await s2iam.get_jwt_database(workspace_group_id="timeout-test-workspace", timeout=30.0)
+    end_time = __import__("time").time()
```

Comment:
What's the `__import__` about? I thought that was only used if you wanted to give a name that wasn't known until runtime.